feat(clerk-react,clerk-js, types,shared): Introduce `Clerk.status` (#5476)
clerk/javascript: .changeset/olive-onions-do.md

---
'@clerk/clerk-js': minor

> '@clerk/clerk-react': minor

Reviewer: Should we remove clerk-react here?

> '@clerk/clerk-js': minor
> '@clerk/clerk-react': minor
> '@clerk/types': minor
> ---

Author: ```suggestion
'@clerk/clerk-js': minor
'@clerk/types': minor
---
```

Introduce `Clerk.status` for tracking the state of the clerk singleton.

> Possible values are:
> - `loading` set on during initialization
> - `error` set when hotloading clerk-js failed or `Clerk.load()` failed
> - `ready` set when Clerk is fully operational
> - `degraded` set when Clerk is partially operational

Reviewer: ```suggestion
Possible values for `Clerk.status` are:
- `"loading"`: Set during initialization
- `"error"`: Set when hotloading clerk-js failed or `Clerk.load()` failed
- `"ready"`: Set when Clerk is fully operational
- `"degraded"`: Set when Clerk is partially operational
```

> `Clerk.loaded` is `true` when `Clerk.status` is either `ready` or `degraded`.
> `Clerk.loaded` is `false` when `Clerk.status` is `loading` or `error`.

Reviewer: ```suggestion
The computed value of `Clerk.loaded` is:
- `true` when `Clerk.status` is either `"ready"` or `"degraded"`.
- `false` when `Clerk.status` is `"loading"` or `"error"`.
```
